tasks/eva_3417/src/main/groovy/eva3417/RemediateLowerCaseRefAlt.groovy

```diff
@@ -0,0 +1,270 @@
+
+                        @Override
+                        void update(ExecutionContext executionContext) throws ItemStreamException {}
+
+                        @Override
+                        void close() throws ItemStreamException {}
+                    })
+                    .writer(svDeprecationWriter)
+                    .listener((StepExecutionListener) dbEnvProgressListener)
+                    .build()
+
+            def svDeprecationJob = dbEnvJobBuilderFactory.get("deprecationJob")
+                    .start(svDeprecateJobSteps)
+                    .incrementer(new RunIdIncrementer()).build()
+            def dbEnvJobLauncher = dbEnv.springApplicationContext.getBean(JobLauncher.class)
+            dbEnvJobLauncher.run(svDeprecationJob, new JobParameters(["executionDate": new JobParameter(LocalDateTime.now().toDate())]))
+        }
+    }
+
+    void updateFileWithHashCollisionList(List<String> sveHashList) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(this.hashCollisionFilePath, true))) {
+            for (String line : sveHashList) {
+                writer.write(line);
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    void updateSVOE(List<SubmittedVariantEntity> impactedSVEList, Map<String, String> mapOldHashNewHash) {
+        List<String> impactedSVEIds = impactedSVEList.stream()
+                .map { sve -> sve.getHashedMessage() }
+                .collect(Collectors.toList())
+        Criteria filterCriteria = new Criteria("inactiveObjects.hashedMessage").in(impactedSVEIds)
+                .and("eventType").is("UPDATED")
+        new RetryableBatchingCursor<SubmittedVariantOperationEntity>(filterCriteria, dbEnv.mongoTemplate, svoeClass).each {
+            sveOperationBatchList ->
+                {
+                    def svoeBulkUpdates = dbEnv.mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED, svoeClass)
+
+                    for (SubmittedVariantOperationEntity svoe : sveOperationBatchList) {
+                        // create update statement
+                        SubmittedVariantInactiveEntity inactiveSVEDocument = svoe.getInactiveObjects()[0]
+                        String oldHash = inactiveSVEDocument.getHashedMessage()
+                        String newRef = inactiveSVEDocument.getReferenceAllele().toUpperCase()
+                        String newAlt = inactiveSVEDocument.getAlternateAllele().toUpperCase()
+                        String newHash = mapOldHashNewHash.get(oldHash)
+
+                        Query findQuery = query(where("inactiveObjects.hashedMessage").is(oldHash))
+                        Update updateQuery = new Update()
+                                .set("inactiveObjects.\$.ref", newRef)
+                                .set("inactiveObjects.\$.alt", newAlt)
+                                .set("inactiveObjects.\$.hashedMessage", newHash)
```

Comment:
I am wondering if there is value in retroactively updating historical ops... #standup